### PR TITLE
move to mobile platform targets to net7

### PR DESCRIPTION
**Pull request by bruno-garcia**
Resolves https://github.com/getsentry/sentry-dotnet/issues/2623

### CODE REVIEW

File: src/Sentry/Internal/Extensions/CollectionsExtensions.cs [hunk 1]
```diff
@@ -29,9 +29,11 @@ internal static class CollectionsExtensions
         }
     }
 
+#if !NET8_0_OR_GREATER
```
Comment (bruno-garcia) [inline src/Sentry/Internal/Extensions/CollectionsExtensions.cs:32]: Technically not needed yet, will be needed when adding .NET 8: https://github.com/getsentry/sentry-dotnet/pull/2617/files#diff-ebf7bc6beaaf9dacfb725d68b7126baccbe823abd28dda757213831c04639774

**Comment by jamescrosswell** [comment]
Need to our sample apps on `net7-ios` (see https://github.com/getsentry/sentry-dotnet/issues/2540#issuecomment-1690794866) and `net8-ios` as well presumably.

File: src/Sentry/Platforms/Android/SentrySdk.cs [hunk 2]
```diff
@@ -220,7 +220,9 @@ private static void AndroidEnvironment_UnhandledExceptionRaiser(object? _, Raise
             return null;
         }
 
+#pragma warning disable CS0618 // Type or member is obsolete
```
Comment (jamescrosswell) [inline src/Sentry/Platforms/Android/SentrySdk.cs:223]: What will be the trigger for when we remove these? When the offending members actually get removed from the dotnet framework?

**Review by jamescrosswell — APPROVED** [review]

**Comment by jamescrosswell** [comment]
All looks good to me... assuming we delay merging until November though (when .NET 8.0 is officially released and we upgrade our tooling)?

**Comment by bruno-garcia** [comment]
> All looks good to me... assuming we delay merging until November though (when .NET 8.0 is officially released and we upgrade our tooling)?

We merge into https://github.com/getsentry/sentry-dotnet/pull/2631 where we collect all breaking changes. We can make preview releases from the branch. In Nov we merge into `main` and release `4.0.0`

**Review by bitsandfoxes — APPROVED** [review]
🔥

**Pull request merged**